src/main/sensors/gyroanalyse.c

```diff
@@ -42,37 +42,37 @@
 #include "sensors/gyroanalyse.h"
 
 // The FFT splits the frequency domain into an number of bins
-// A sampling frequency of 1000 and max frequency of 500 at a window size of 32 gives 16 frequency bins each with a width 31.25Hz
+// A sampling frequency of 1000 and max frequency of 500 at a window size of 32 gives 16 frequency bins each 31.25Hz wide
 // Eg [0,31), [31,62), [62, 93) etc
 
+// for gyro loop >= 4KHz, sample rate 2000 defines to 1000Hz, 16 bins each 62.5 Hz wide
+// NB  FFT_WINDOW_SIZE is defined as 32 in gyroanalyse.h
 #define FFT_BIN_COUNT             (FFT_WINDOW_SIZE / 2)
-// for gyro loop >= 4KHz, analyse up to 666Hz, 16 bins each 41.625 Hz wide
-#define FFT_SAMPLING_RATE_HZ      1333
-// following bin must be at least 2 times previous to indicate start of peak
-#define FFT_MIN_BIN_RISE          2
-// the desired approimate lower frequency when calculating bin offset
-#define FFT_BIN_OFFSET_DESIRED_HZ 90
-// lowpass frequency for smoothing notch centre point
-#define DYN_NOTCH_SMOOTH_FREQ_HZ  60
-// notch centre point will not go below this, must be greater than cutoff, mid of bottom bin
-#define DYN_NOTCH_MIN_CENTRE_HZ   125
+// the number of bins to ignore in FFT analysis (1 is the minimum)
+#define FFT_BIN_OFFSET            1
+#define DYN_NOTCH_SMOOTH_FREQ_HZ  50
+// notch centre point will not go below sample rate divided by these dividers, resulting in range limits:
+// HIGH : 166-1000Hz, MEDIUM -> 111-666Hz, LOW -> 83-500Hz
+#define DYN_NOTCH_MIN_CENTRE_DIV  12
 // lowest allowed notch cutoff frequency
-#define DYN_NOTCH_MIN_CUTOFF_HZ   105
+#define DYN_NOTCH_MIN_CUTOFF_DIV  10
 // we need 4 steps for each axis
 #define DYN_NOTCH_CALC_TICKS      (XYZ_AXIS_COUNT * 4)
 
-static uint16_t FAST_RAM_ZERO_INIT fftSamplingRateHz;
-// centre frequency of bandpass that constrains input to FFT
-static uint16_t FAST_RAM_ZERO_INIT fftBpfHz;
-// Hz per bin
-static float FAST_RAM_ZERO_INIT    fftResolution;
-// maximum notch centre frequency limited by Nyquist
-static uint16_t FAST_RAM_ZERO_INIT dynNotchMaxCentreHz;
-static uint8_t  FAST_RAM_ZERO_INIT fftBinOffset;
+static uint16_t FAST_RAM_ZERO_INIT   fftSamplingRateHz;
+static float FAST_RAM_ZERO_INIT      fftResolution;
+static uint8_t FAST_RAM_ZERO_INIT    fftBinOffset;
+static uint16_t FAST_RAM_ZERO_INIT   fftBpfHz;
+static uint16_t FAST_RAM_ZERO_INIT   dynamicNotchMinCenterHz;
+static uint16_t FAST_RAM_ZERO_INIT   dynamicNotchMaxCenterHz;
+static uint16_t FAST_RAM_ZERO_INIT   dynamicNotchMinCutoffHz;
+static float FAST_RAM_ZERO_INIT      dynamicFilterWidthFactor;
+static uint8_t FAST_RAM_ZERO_INIT    dynamicFilterType;
+
+static uint8_t dynamicFilterRpmRange;
```

Comment:
RPM? Semantics of this look wrong to me

Reply:
I agree, it's an awkward term.  
I was meaning to imply to the user that they should select LOW, MED and HIGH on the basis of expected RPM. 
But it wasn't a very good name, to be honest I was a bit stuck for the name.
What names do you think make the most sense?

Reply:
Removed RPM